# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
# Commit: dea9f12cdec0
# Full Hash: dea9f12cdec02aeb5ac260a970659e09dc73a375
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-07-02 09:15:46
# Regressor Bug: 1560936
# File Overlap Count: 3
# Description:
#   Bug 1560936 - For getSettings make all MediaStreamTrackSources populate width and height as appropriate. r=jib,bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251430
# ==============================================================================

diff -r a4ae1edb514a -r dea9f12cdec0 dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
--- a/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Tue Jul 01 18:42:42 2025 +0000
+++ b/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Tue Jul 01 18:42:43 2025 +0000
@@ -68,13 +68,6 @@
 
  public:
   /**
-   * Callback Function reportng any change in the video-frame dimensions
-   * @param width:  current width of the video @ decoder
-   * @param height: current height of the video @ decoder
-   */
-  virtual void FrameSizeChange(unsigned int width, unsigned int height) = 0;
-
-  /**
    * Callback Function reporting decoded frame for processing.
    * @param video_frame: reference to decoded video frame
    * NOTE: If decoded video frame is passed through buffer , it is the
@@ -397,6 +390,7 @@
   virtual Maybe<Ssrc> GetAssociatedLocalRtxSSRC(Ssrc aSsrc) const = 0;
 
   virtual Maybe<gfx::IntSize> GetLastResolution() const = 0;
+  virtual AbstractCanonical<Maybe<gfx::IntSize>>* CanonicalReceivingSize() = 0;
 
   virtual void RequestKeyFrame(FrameTransformerProxy* aProxy) = 0;
   virtual void GenerateKeyFrame(const Maybe<std::string>& aRid,